library/core/src/fmt/num.rs

```diff
@@ -195,8 +198,30 @@ macro_rules! debug {
     )*};
 }
 debug! {
-  i8 i16 i32 i64 i128 isize
-  u8 u16 u32 u64 u128 usize
+  i8 i16 i32 i64 #[cfg(not(no_128_bit))] i128 isize
+  u8 u16 u32 u64 #[cfg(not(no_128_bit))] u128 usize
+}
+
+macro_rules! fake_debug {
+    ($($( #[$cfg:meta] )? $T:ident)*) => {$(
+        $( #[$cfg] )?
+        #[stable(feature = "rust1", since = "1.0.0")]
+        impl fmt::Debug for $T {
+            #[inline]
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                // Debug formats are not stable so this is a legitimate implementation.
+                // It would be possible to hexdump the contents instead, or to
+                // fail or panic, but for an application which has specifically
+                // recompiled core to avoid i128, the former seems unnecessary
+                // and the latter needlessly harmful.
+                f.write_str(stringify!($T))
```

Comment:
Why is an i128/u128 impl necessary in the first place?

Reply:
Good question; I haven’t tried removing it. I think it’s conventional to provide `Debug` for everything public in the standard library; not sure if that could be waived in this case?